content/docs/1.5.4/high-availability/auto-remount-volume-by-deleting-workload-pod.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Auto Remount Volume By Deleting Workload Pod
+weight: 1
+---
+
+When the following situations occur, Longhorn automatically attempts to delete workload pods that are managed by a controller (for example, Deployment, StatefulSet, or DaemonSet). After deletion, the controller restarts the workload pod and Kubernetes handles volume reattachment and remounting.
+
+1. A volume was unexpectedly detached, possibly because of a [Kubernetes upgrade](https://github.com/longhorn/longhorn/issues/703), [Docker reboot](https://github.com/longhorn/longhorn/issues/686), network connectivity issue, or volume engine crash.
+2. A volume was automatically salvaged after all replicas became faulty, possibly because of a network connectivity issue. Longhorn attempts to identify the usable replicas and uses them for the volume.
+3. An error occurred on a Share Manager pod that uses an RWX volume.
```

Comment:
Item 3 is incorrect. Consumer workloads should not be deleted when a share manager pod encounters an error and is subsequently restarted.

https://longhorn.io/docs/1.6.0/references/settings/#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly

Reply:
We do delete the user workload pod when the share-manager pod encounters an error and restarted
Here is the code, and I have tested it before, it did kill and restart the user pod.
https://github.com/longhorn/longhorn-manager/blob/edf23eddc3b6c307031eaa770c3d312c963f25a5/controller/volume_controller.go#L4153-L4161

cc @ejweber @shuo-wu

Reply:
@derekbit @james-munson  does this mean share manager failover only takes effect when users disable `auto-delete-pod-when-volume-detached-unexpectedly`?? If it's enabled (default), the consumer workloads are always deleted to restart when the share manager pod is stopped somehow.

If yes, should we consider a different handling for RWX to make the failover happen w/o interrupting the consumer workloads by default w/o disabling the global setting?

Reply:
@james-munson James mentioned that there is a long history for this part. It seems that we disallowed then re-allowed setting `auto-delete-pod-when-volume-detached-unexpectedly` to handle RWX volumes. But I don't know why...
Currently, Yes, once the setting is enabled, `workloads are always deleted to restart when the share manager pod is stopped somehow`